### PR TITLE
Feature/add device2bundlechanges

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

*Please replace with a description about why this change is needed?*

added `Device` support for `bundle` `facade`.

## QA steps

*Please replace with how we can verify that the change works?*

Client only, needs back-end support

## Documentation changes

It only includes client side changes for device support on bundle deploy.
Document changes will be required once back-end done.

### CODE REVIEW

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
!!build!!

**Comment by ycliuhw** [comment]
!!build!!

File: testcharms/charm-repo/bundle/wordpress-with-mysql-devices/bundle.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+services:
```
Comment (wallyworld) [inline testcharms/charm-repo/bundle/wordpress-with-mysql-devices/bundle.yaml]: I think we should invent a new better charm/bundle name for using devices.

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
[add ability of parsing device constraints to bundlechanges repo](https://trello.com/c/uCvdH2xt/298-add-ability-of-parsing-device-constraints-to-bundlechanges-repo)

**Pull request merged**